delete_expired_objects/delete_expired_objects_job.py

```diff
@@ -1,32 +1,49 @@
 from time import time
 from gen3 import object, metadata, auth
 
+"""
+Note: This job requires Metadata service to be above version 1.6.4
+"""
 auth = auth.Gen3Auth(refresh_file="/mnt/api-profile-credentials.json")
 mds_handle = metadata.Gen3Metadata(auth)
 
-print("Initializing delete_temp_objects_job...")
-LIMIT_SIZE = 10
+print("Initializing delete_expired_objects_job...")
+
+LIMIT_SIZE = 2000
 offset_position = 0
 response_dict = mds_handle.query(
     query="date_to_delete=*", return_full_metadata=True, limit=LIMIT_SIZE
 )
-if type(response_dict) is dict:
-    guid_list = list(response_dict.values())
-    while len(response_dict) == LIMIT_SIZE:
-        offset_position += LIMIT_SIZE
-        response_dict = mds_handle.query(
-            query="date_to_delete=*",
-            return_full_metadata=True,
-            limit=LIMIT_SIZE,
-            offset=offset_position,
-        )
-        guid_list += list(response_dict.values())
-    delete_list = [obj["guid"] for obj in guid_list if obj["date_to_delete"] < time()]
-    object_api = object.Gen3Object(auth)
-    for guid in delete_list:
+
+if type(response_dict) is not dict:
+    raise Exception(f"Job Failed. Couldn't fetch records. -- {response_dict}")
+
+
+guid_list = list(response_dict.values())
+while len(response_dict) == LIMIT_SIZE:
+    offset_position += LIMIT_SIZE
+    response_dict = mds_handle.query(
+        query="date_to_delete=*",
+        return_full_metadata=True,
+        limit=LIMIT_SIZE,
+        offset=offset_position,
+    )
+    guid_list += list(response_dict.values())
```

Comment:
> You are keeping a lot of data in memory by constructing the whole guid_list first, and by storing all the metadata instead of only what you need (guid and date_to_delete).

can we reduce the data we store in `guid_list` to only what we need?

--
edit: maybe we can even remove the ones we don't need at this point instead of later, to reduce the amount of data we store and have a shorter list to iterate through when we delete? `guid_list += [e["guid"] for e in response_dict.values()] if e["date_to_delete"] < now]`

^ i think this will be more performant than the current code, because by doing `list(response_dict.values())` you are already iterating through the list anyway